src/pudl/transform/ferc1.py

```diff
@@ -806,6 +809,37 @@ def transform(
         return df
 
 
+class PlantsHydroFerc1TableTransformer(Ferc1AbstractTableTransformer):
+    """A table transformer specific to the ``plants_hydro_ferc1`` table."""
+
+    table_id: Ferc1TableId = Ferc1TableId.PLANTS_HYDRO_FERC1
+
+    @cache_df(key="main")
+    def transform_main(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Table specific transforms for plants_hydro_ferc1. Hi!
+
+        Params:
+            df: Pre-processed, concatenated XBRL and DBF data.
+
+        Returns:
+            A single transformed table concatenating multiple years of cleaned data
+            derived from the raw DBF and/or XBRL inputs.
+        """
+        df = (
+            self.normalize_strings(df)
+            .pipe(self.categorize_strings)
+            .pipe(self.convert_units)
+            .pipe(self.nullify_outliers)
+            .pipe(self.drop_invalid_rows)
+        )
```

Comment:
If you define all of the paremeters for these functions in a dictionary in the `params` module, I don't see why we have to list them out here instead of just calling on the params dictionary with a function call as a key and iterate through it. It is nice to see all of the steps outlined in this function, but does it make more sense to just have the transform function run through the params dictionary? Lmk if this doesn't make sense.

Reply:
That does make sense conceptually! I could see that being very nice as a general design, but the transformers are not that smart. We had talked about doing this, but the complexities of building a transformer that would know how to iterate through the params with the correct order and with the correct inputs (i.e. does `categorize_strings` always take the post-concat-ed df? how does the `process_xbrl` get both the instant and duration tables while `process_dbf` just gets one?) were just a little too much. Maybe one day though!

Reply:
There's too much variety across the different tables, so the exact ordering and any special steps that only belong to a specific table and aren't parameterized have to be run as appropriate in here. That's pretty much the role of these transform methods -- composing and ordering the transformations as needed for this particular table.

Reply:
ok gocha